Gordon360/Services/LostAndFoundService.cs

```diff
@@ -148,44 +153,83 @@ public int CreateActionTaken(int id, ActionsTakenViewModel ActionsTaken)
             }
             else
             {
-                throw new ResourceCreationException() { ExceptionMessage = "No account could be found for the user." };
+                throw new ResourceCreationException() { ExceptionMessage = "No account could be found for the requesting user." };
             }
 
-            var newActionTaken = context.ActionsTaken.Add(new ActionsTaken
+            var missingItemReport = context.MissingReports.Find(missingItemID);
+            if (missingItemReport == null)
             {
-                missingID = id,
-                action = ActionsTaken.action,
-                actionNote = ActionsTaken.actionNote,
-                actionDate = ActionsTaken.actionDate,
-                submitterID = idNum,
-                isPublic = ActionsTaken.isPublic,
-            });
-
-            context.SaveChanges();
+                throw new ResourceNotFoundException() { ExceptionMessage = "No missing item report with given id found in the system" };
+            }
 
-            if (newActionTaken == null || newActionTaken?.Entity?.ID == 0)
+            // If the report belongs to the requesting user, or the user is an admin
+            if (missingItemReport.submitterID == idNum || hasFullPermissions(username))
```

Comment:
This would traditionally be handled in the StateYourBusiness logic. I'm not sure if it's actually better there or not, but thought I'd at least mention it.

Reply:
I think I'll leave this as is at this point, I don't see a huge argument to do it in stateYourBusiness rather than here, especially for internal consistency in our routes, since some of the business logic for authorization in the routes has several conditions.